### PR TITLE
feat(conftest): Added root level conftest file

### PR DESCRIPTION
**Pull request by hrushikesh430**
Added the conftest.py file at root level of repo, the fixture and configuratuions can be used all over repo.

# Description

- Added the conftest.py file at the root level. Added the config and build_dir fixture in the conftest.py. 
- Tested the file using in pre_encrypted_ota example pytest ([PR](https://github.com/espressif/idf-extra-components/pull/398)).

### CODE REVIEW

**Comment by hrushikesh430** [comment]
@mahavirj @hfudev PTAL

**Review by hfudev — APPROVED** [review]
LGTM.

File: conftest.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# SPDX-FileCopyrightText: 2021-2024 Espressif Systems (Shanghai) CO LTD
```
Comment (mahavirj) [inline conftest.py:1]:
```suggestion
# SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
```

**Comment by mahavirj** [comment]
@hfudev What could be this CI failure here: https://github.com/espressif/idf-extra-components/actions/runs/12410469543/job/34646649976?pr=455?

**Comment by hrushikesh430** [comment]
Right now, this PR is not needed. I will reopen after multiple CI configuration build is supported in this repository.

**Pull request closed without merging**